Proofread, reorganized, and added to all the "Python on Windows" guides (#794)
discord-python/site: pydis_site/apps/content/resources/guides/python-guides/windows/installing-and-using-python.md

---
title: Installing and Using Python on Windows
description: How we recommend installing Python on Windows, and how to use Python Windows features
icon: fab fa-windows
toc: 3
---

Our recommended way of installing Python on a Windows operating system is using the full installer from the official

> [python.org Downloads page](https://www.python.org/downloads/) (hit the big yellow button) using the default options,

Reviewer: ```suggestion
[python.org Downloads page](https://www.python.org/downloads/) using the default options,
```

Maybe better to force them to read the full instructions, only stressing adding to path here? Pretty minor though, fine how it is

Author: I'll "hit" to "from". Just like to make it clear for those who often only read the first paragraph of a tutorial (i.e. often me).

except making sure to check the "Add python.exe to PATH" checkbox. Getting Python from the [Microsoft
Store](https://apps.microsoft.com/store/search/python) is _not_ recommended as [it can cause various
issues](../microsoft-store).

This guide gives detailed instructions on that [recommended way to install Python on Windows](#recommended-install),
then goes on to discuss [more information about installing Python on Windows](#more-installation-information), and
finally wraps up by explaining some common Windows-specific Python usage, namely the [py launcher](#the-py-launcher),
and how to work with [virtual environments](#virtual-environments).

You may also want to check out our guides on [Common Issues Using Python on Windows](../common-issues), [Adding Python
to the Windows Path](../putting-python-on-path.md), and [Setting up a Unix-Style Environment on
Windows](../unix-env-on-windows.md).

## Recommended Install

Follow the five steps below to install the latest version of Python on Windows.

> (The instructions were written with Windows 10 and Python 3.11.0 in mind, but should be nearly or fully identical with

Reviewer: If it's the same with windows 11 I'm not sure we need to mention this.

Author: I'd prefer to mention it. Lots of times I see tutorials that don't mention the software versions involved or date of writing and I have to guess if it'll work for my setup.

Mentioning Win 11 might help people from clicking away thinking "oh this must be a Win 10 guide since Win 11 is new".

(Though btw I have not explicitly tested the guides in Win 11 since I'm still on Win 10.)

Reviewer: Fair, fwiw i'm on windows 11 and everything seems to be the same.

Windows 11 or other modern versions of Python.)

> > If you want a fresh start, you may want to first uninstall any other versions of Python on your PC, including those

Reviewer: Maybe "If you want a fresh start, you should check for and uninstall versions of Python already on your PC", to encourage people to check even if they think they don't have anything installed.

Author: I like this wording better too. Changing it.

> from the Microsoft Store, if you have any. This can be done in the ["Apps & features" Windows
> settings](/static/images/content/python-on-windows/ms_store_uninstall.png) (type "apps and features" into the Start
> Menu to find it). Though it's fine to have multiple versions of Python installed at once. It can be useful for testing

> > version compatibility or for working on projects made in a certain version. Only uninstall things if you want to.

Reviewer: Starting this sentence with "Though" reads a bit weirdly, maybe that's just me though?

Maybe "It's also fine ... which can be useful ...", or "That said, it's also fine ... which can be useful ..."?

1.  Go to [python.org/downloads](https://www.python.org/downloads) and click the big yellow "Download Python 3.x.x"
    button near the top of the page. That should start the download of the latest Windows Python installer that best

>     suits your computer.

Reviewer: ```suggestion
    button near the top of the page. That should start the download of the installer for the latest version that best
    suits your computer.
```
Not sure, was trying to make it so the "latest" is referring to the python version, but i'm not sure it's more clear. Thoughts?

    If you want a different version or it doesn't work for some reason, you can download the
    Windows installer you want from [python.org/downloads/windows](https://www.python.org/downloads/windows).

    [![Step 1](/static/images/content/python-on-windows/recommended_install_1.png)](/static/images/content/python-on-windows/recommended_install_1.png)

2.  When it finishes downloading, click the file in your browser or find it in your Downloads folder and double-click it
    to start the installer.

    [![Step 2](/static/images/content/python-on-windows/recommended_install_2.png)](/static/images/content/python-on-windows/recommended_install_2.png)

3.  Check the "Add python.exe to PATH" checkbox (the text may differ slightly depending on your installer). This will
    make it so terminals can recognize commands like `python` and `pip`. (Read [this
    guide](../putting-python-on-path) to learn more.)

    [![Step 3](/static/images/content/python-on-windows/recommended_install_3.png)](/static/images/content/python-on-windows/recommended_install_3.png)

4.  Then click the big "Install Now" button. Aside from adding Python to PATH, the rest of the installer defaults are

>     usually fine, so there's no need to customize the installation unless you want to.

Reviewer: ```suggestion
    usually fine, so there's no need to customize the installation unless you have a specific reason to.
```
"want to" is a bit vague - I would steer people away from customising the installation unless they have a reason to.

    [![Step 4](/static/images/content/python-on-windows/recommended_install_4.png)](/static/images/content/python-on-windows/recommended_install_4.png)

5.  It will take a minute to install and then (hopefully) say "Setup was successful".

    Congrats! You just installed Python! You can close the installer.

    [![Step 5 A](/static/images/content/python-on-windows/recommended_install_5.png)](/static/images/content/python-on-windows/recommended_install_5.png)
    [![Step 5 B](/static/images/content/python-on-windows/recommended_install_6.png)](/static/images/content/python-on-windows/recommended_install_6.png)

### Checking that it Worked

To test that installing Python worked, you can do what it suggests and search "python" on the Start Menu to find the
Python console app and run some code like `print("Hello, World!")`.

[![Testing Python console](/static/images/content/python-on-windows/recommended_install_7.png)](/static/images/content/python-on-windows/recommended_install_7.png)

Or try the more usual way of running Python by typing `python` in a new terminal window to open up the Python

> [REPL](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop), or use `python somefile.py` to run some

Reviewer: ```suggestion
[REPL](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop), or use `python somefile.py` to run a
```

Python file. You can use whichever terminal you prefer: Command Prompt, PowerShell, an IDE-integrated terminal, [Windows
Terminal](https://apps.microsoft.com/store/detail/windows-terminal/9N0DX20HK701), etc. It just has to be a freshly
opened terminal or the commands may not be recognized.

[![Testing py and python](/static/images/content/python-on-windows/recommended_install_8.png)](/static/images/content/python-on-windows/recommended_install_8.png)

You can double check what versions of Python and [pip](https://pip.pypa.io/en/stable/) were installed by running
`python -V` or `pip -V` in a terminal:

[![Checking python and pip versions](/static/images/content/python-on-windows/testing_path_worked_1.png)](/static/images/content/python-on-windows/testing_path_worked_1.png)

And finally, it's worth searching and opening "IDLE" on the Start Menu, which is the the basic
[IDE](https://en.wikipedia.org/wiki/Integrated_development_environment) that comes with Python. Most people end up using
[PyCharm](https://www.jetbrains.com/pycharm/) or [VSCode](https://code.visualstudio.com/) to program in Python but IDLE
is a great starting point.

> [![Testing IDLE](/static/images/content/python-on-windows/recommended_install_9.png)](/static/images/content/python-on-windows/recommended_install_9.png)

Reviewer: I think the topic of editors deserves it's own page, as it's hard to get across much info in a small space here. Not something for this PR though so I guess this is OK for now.

## More Installation Information

### Selecting an Installer

At [python.org/downloads/windows](https://www.python.org/downloads/windows/) there are many different Windows Python
installer options available. A link to the latest release page is listed at the top, for example [the Python 3.11.0
release page](https://www.python.org/downloads/release/python-3110/) has info about the new 3.11 features, with
downloads at the bottom. Back on the list of all Windows releases, pre-releases are on the right if you want to try

> cutting-edge (and possibly unstable) versions of Python, and stables releases are on the left, all the way back to
> Python 2.0.1 from 2001. Some minor release versions may not have installers available.

Reviewer: ```suggestion
cutting-edge (and possibly unstable) versions of Python, and stables releases are on the left. 
Some minor release versions may not have installers available.
```
Feel free to ignore, not sure this adds anything though.

Author: Eh, I like the little context of showing how old Python is.

on the Path so you can invoke it with `python`, and then using `py -X.Y` whenever you want a different version, e.g.
`py -3.7`.

The official documentation of the `py launcher` can be found
[here](https://docs.python.org/3/using/windows.html#python-launcher-for-windows). Use `py --help` to show help on the
command line.

The py launcher is not available on GNU/Linux-based platforms such as macOS, but there is [an unofficial version by
Brett Cannon](https://github.com/brettcannon/python-launcher).

## Virtual Environments

Virtual environments ("venvs") are a way of giving each of your Python projects their own space to install dependencies
in, so different projects can have different versions of the same dependencies. For example, suppose you are working on
two websites that both use the Python web-framework [Django](https://www.djangoproject.com/download/). One site is older
and requires Django 3.2, and the other is newer and requires Django 4.1. If you install Django globally, no matter if
you install 3.2 or 4.1, one or the or the other of the projects will have the wrong version. But if you make a virtual

> environment for each project you can install the required Django version for each separately.

Reviewer: ```suggestion
you install 3.2 or 4.1, one or the or the other of the projects will have the wrong version. If you make a virtual
environment for each project you can install the required Django version for each separately.
```
or "To solve this, you can make a virtual environment for each project and install the required Django version for each separately.", just starting a sentence with "But" isn't great.

Author: I'll change it to "If instead...". (Though I definitely overuse though transition words in my writing style :P)

> Virtual environments are not exclusive to Windows, but the commands to use them can differ a bit across operating

Reviewer: ```suggestion
Virtual environments are not exclusive to Windows, but the commands to use them can differ across operating
```
not really necessary
